docs(legend): some legends and lore

AleksueiR opened the pull request:
## Description
Magnum opus on legend (lots of typos I'm sure).

Closes #1090, #1091

## Testing
Nope.

## Documentation
It is that.

## Checklist
<!-- Quick checklist for items that are easy to miss -->

- [x] Commit messages follow [the guidelines](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/master/CONTRIBUTING.md#-git-commit-guidelines)
- [ ] Release notes have been updated
- [x] PR targets the correct release version
- [x] Help files and documentation have been updated

Remember, it is a *muffin offence* to open a PR with any of the above checklist items incomplete.

Please keep the original issue up to date with the final solution, expected behaviour, and any additional notes for testers

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/2355)
<!-- Reviewable:end -->

james-rae commented:
Reviewed 1 of 3 files at r1.
Review status: 1 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 2 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Generated docs are here: https://fgpv-vpgf.github.io/fgpv-vpgf/docs-legend-n-lore/tutorial-legend.html

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-:-KtM78TOwwrGKMNPTwWl:b-5q1wqc)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 22 unresolved discussions, all commit checks successful.

---

*[docs/developer/technical/legend.md, line 1 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMaw0xIDLge8fYIaQY:-KtMaw0xIDLge8fYIaQZ:bmi1h3f) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L1)):*
> ```Markdown
> Let's start by making everything more confusing that it was a moment ago. First, some terminology:
> ```

I would leave a general note somewhere on this page as to what version of the schema was used for all the example code (could be in a comment or somewhere in the footer).  Should make it easier when docs need to be updated to a new version of the schema.

---

*[docs/developer/technical/legend.md, line 5 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMN99VDZlnzLjdMfZJ:-KtMN99WzVpc8FSw38BI:b-9cziic) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L5)):*
> ```Markdown
> ## Terminology
> ### Legend as Devs see it
> __Legend__ is a list of (legend) blocks displayed in the main panel (toc pane - table of content). When opened, the legend pane is titled "Layers" - due to popular user demand.
> ```

table of contents

---

*[docs/developer/technical/legend.md, line 12 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMNEtvgD29eI7ytQE8:-KtMNEtvgD29eI7ytQE9:b-ommp3a) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L12)):*
> ```Markdown
> 
> > Devs call it: legend, layer Selector, layer List
> > Users call it: I'm not sure what _users_ call this.
> ```

formatting loses the line break

---

*[docs/developer/technical/legend.md, line 21 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMN_blNzQTk8OEZOhu:-KtMN_blNzQTk8OEZOhv:b-tcqycs) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L21)):*
> ```Markdown
> Symbology stacks/legends can display custom images specified in the config and be added to the legend blocks not connected to actual layers.
> 
> > Devs call it: symbology stack
> ```

This is repeated from above

---

*[docs/developer/technical/legend.md, line 30 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMNmsE9kV9vX81qr7U:-KtMNmsE9kV9vX81qr7V:b-3hp9z0) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L30)):*
> ```Markdown
> The legend comes in two flavours - ~~vanilla and chocolate~~ structured and autogenerated.
> 
> __Structured__ legend lets the config author to mess with the ordering of legend blocks, their names, specify custom symbology stacks, put things into regular groups or visibility sets, hide layer from the legend, and have a single legend block controlling several layers. Lots of power options, plenty of opportunities to screw up the config. All this should be used for thematic maps, maps that tell a certain story with the data.
> ```

remove the `to` after config author
`hide layer` -> `hide layers`
I would rewrite the last sentence as: All this should be used for thematic maps (i.e. maps that tell a certain story with the data).

---

*[docs/developer/technical/legend.md, line 34 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMOzfXqXaipcXM4-gp:-KtMOzfXqXaipcXM4-gq:bya81ku) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L34)):*
> ```Markdown
> __Autogenerated__ legend is simple - it takes the layer list and turns it into a legend, one to one correspondence (one regular legend block per one layer).
> 
> It should be pointed out the __autogenerated__ legend - after it's generated during the config parse - is technically a __structured__ legend. It uses a subset of elements which can be used in the __structured__ legend.
> ```

`after it's` ->`after it has`

---

*[docs/developer/technical/legend.md, line 87 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMbM93VKoGBt4dukOK:-KtMbM93VKoGBt4dukOL:bi4l2il) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L87)):*
> ```Markdown
> #### Info Section
> 
> __Info section__ is a piece of static content (unless it's a .gif) which can be added between any other legend blocks. There are four types:
> ```

I would add all four types here, then have the full description sections below (so `There are four types: text, title, image, and unbound layer`).

---

*[docs/developer/technical/legend.md, line 178 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMbzzMYBMDNoOdoqN0:-KtMbzzMYBMDNoOdoqN1:b-zcgrx1) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L178)):*
> ```Markdown
> }
> ```
> ```

I would give a slightly more detailed example and explain the more important properties (maybe something with layerId, entryIndex, symbologyStack and symbologyRenderStyle).  Something people can easily copy and paste to get started.

---

*[docs/developer/technical/legend.md, line 224 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMgZVT1g8pc2U7rsgF:-KtMgZVT1g8pc2U7rsgG:b-nb24hc) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L224)):*
> ```Markdown
> #### Entry Group (config) or Legend Group (code)
> 
> These let you group things into collapsible sections. Anything can be put inside a group, other groups including (group nesting). Nesting them more than ten levels deep will not produce any changes in the UI - it's a flat list from there on.
> ```

either `other groups included` or `including other groups`

---

*[docs/developer/technical/legend.md, line 255 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMh_KGMzqQpsG7xboU:-KtMh_KGMzqQpsG7xboV:b-pd5jev) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L255)):*
> ```Markdown
> #### Visibility Set
> 
> This is a convenience for the user when at most a single element from a set can be visible at a time - it's possible to have all elements in the set turned off. They groups things together, but there is no header element. It's also not possible to nest visibility sets directly - it just doesn't make sense. You can put a set into a group and put that group into another set, if that's your fancy. Trying to make nested sets will only confuse the users and devs.
> ```

Maybe provide a bit more guidance to users on when they should use a visibility set vs normal layers (giving a few good and bad use cases).

---

*[docs/developer/technical/legend.md, line 343 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMi9mLEJW9BMby3kPC:-KtMi9mLEJW9BMby3kPD:b5y9oa6) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L343)):*
> ```Markdown
> We don't know what will happen, maybe a black hole will form. Don't do this.
> 
> As a rule of thumb, if it doesn't make logical sense, it probably won't work even if the config validates.
> ```

On the dev side we should explain that trying to catch all these cases would result in a hugely bloated schema that no one would want to ever touch.  A plea for config authors to be sensible seemed much less painful.

---

*[docs/developer/technical/legend.md, line 357 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMj-tx2qU9OyIx6IKj:-KtMj-tx2qU9OyIx6IKk:b-d0eg97) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L357)):*
> ```Markdown
> | x       | x               | control is visible and is locked to modifications                              |
> 
> ## Legend Generation (LegendService)
> ```

This is probably where the user facing side of the docs end.  Maybe add another top level break or have some form of note for the users?

---

*[docs/developer/technical/legend.md, line 359 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMjIci9l0UHzwkQvlY:-KtMjIcjMUm4QRvgL0Gc:b-9npc05) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L359)):*
> ```Markdown
> ## Legend Generation (LegendService)
> 
> The LegendService is responsible for converting the entries, entryGroups, and other items from the legend section of the config into appropriate LegendBlocks which will then be rendered in UI. The concept is simple - take an legend entry from the config, create a LegendBlock, and link it to the corresponding LayerRecord, and yet the `legend.service.js` file is 800 lines long.
> ```

is *currently* 800 lines :)

---

*[docs/developer/technical/legend.md, line 363 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMjljMhLjkknrHXkfW:-KtMjljMhLjkknrHXkfX:bqyfcod) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L363)):*
> ```Markdown
> ### constructLegend
> 
> This function is invoked only once per every config load and will generate initial legend based on the config data. All additional layers need to be added to the legend separately, one by one. `constructLegend` function takes in an array of legend block and an array of layer definitions. Each layer definition is converted into a layer blueprint for future references by the legend blocks. It's possible for a single layer record to be referenced in several layer blocks, and for some layer records not to be used at all. Layer blueprints are just that, dormant layer record templates which are only instantiated when a corresponding reference is found in a legend block.
> ```

`legend block` -> `legend blocks`

---

*[docs/developer/technical/legend.md, line 365 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMk2091cwaEJtXQGiC:-KtMk2091cwaEJtXQGiD:b3a0m92) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L365)):*
> ```Markdown
> This function is invoked only once per every config load and will generate initial legend based on the config data. All additional layers need to be added to the legend separately, one by one. `constructLegend` function takes in an array of legend block and an array of layer definitions. Each layer definition is converted into a layer blueprint for future references by the legend blocks. It's possible for a single layer record to be referenced in several layer blocks, and for some layer records not to be used at all. Layer blueprints are just that, dormant layer record templates which are only instantiated when a corresponding reference is found in a legend block.
> 
> Next, the legend tree is processed. Group, Info, and Set legend blocks are very straightforward to create - they are static and not directly connected to the layer record. Groups and Sets are containers for other blocks, while Into blocks are just hold some text or image data.
> ```

`Into` -> `Info`

---

*[docs/developer/technical/legend.md, line 371 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMkKKq8siZpZ6BxxWn:-KtMkKKrs8Y7nbdLvlMv:bvog9xg) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L371)):*
> ```Markdown
> This is used when reloading or removing legend blocks. Since the user can issue to reload or remove a single legend block, and several legend blocks can potentially reference a single layer, all legend blocks connected to that layer need to be re-created. This mapping makes it easier to track such cases.
> 
> When creating a Legend Node block (`_makeLegendBlock`), first the appropriate blueprints are found - there must be a single main blueprint and zero or move controlled blueprints (`controlledIds` option from the config). Then, legend block is mapped to the entry type:
> ```

`zero or move` -> `zero or more`

---

*[docs/developer/technical/legend.md, line 375 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMk_n_yeu4v9MKK73F:-KtMk_n_yeu4v9MKK73G:b-clq5ip) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L375)):*
> ```Markdown
> - all single entries whose main blueprint is of `esriDynamic` type are rendered as `LegendGroup`s
> - all child entries of a entry referencing an `EsriDynamic` layer, are rendered according to that layer's structure, after the layer has been loaded - this cannot be determined upfront
> - `ogcWMS` are rendered as `LegendNode`s for now
> ```

I would make this a sub point of the first bullet since it is covered by that case and just additional info

---

*[docs/developer/technical/legend.md, line 379 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMl3Hz1YYGURw4qiLx:-KtMl3Hz1YYGURw4qiLy:b5w43ph) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L379)):*
> ```Markdown
> #### _makeNodeBlock
> 
> First, a referenced layer record for this block is retrieved from the `layerRegistry` - the record might already exist or not (if not, it's created and loaded), and its proxy object extracted. This happens in `_getLegendBlockProxy`.
> ```

Maybe include a bit about what the proxy object is or a link to the relevant documentation.

---

*[docs/developer/technical/legend.md, line 432 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMmoc8UJ4qvwrxRuWn:-KtMmoc8UJ4qvwrxRuWo:bmprhsa) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L432)):*
> ```Markdown
> First, we get the `ProxyWrapper` for the Dynamic layer itself and construct a new `LegendGroup` config for it. A new group config is needed since an `entry` is turned into a `group` and groups have some different controls and settings. With the new config and the `ProxyWrapper` the `LegendGroup` block is created and mapped to its layer. Then a on-load listener is set which will run `_createDynamicChildTree` to create a tree since now we know the layer structure. The tree is parsed and all its children are added to the `LegendGroup` block created in the beginning (`_addChildBlock`). The settings of leaves are synchronized (`LegendNode.applyInitialStateSettings`).
> 
> The controlled layers are processed in a similar passion to controlled layers on a `LegendNode` with huge difference - controlled `ProxyWrapper`s are actually turned into normal legend blocks which are added to the group as usual. These blocks are marked as `controlled` which precludes them from being rendered.
> ```

`passion` -> `fashion`?

---

*[docs/developer/technical/legend.md, line 436 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMnJuiCDJjbSWqcLCc:-KtMnJuiCDJjbSWqcLCd:blftlls) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L436)):*
> ```Markdown
> #### _getControlledLegendBlockProxy
> 
> This function is used by all layer types to create a flat list of controlled `ProxyWrapper`s. First, controlled wrappers get their `boundingBox` and `query` settings and states disabled. If the controlled layer is not a Dynamic one, layer record is retrieved, its proxy object is obtained, wrapped and returned. Easy.
> ```

`If the controlled layer is not a Dynamic one: the layer record is retrieved, its proxy object is obtained, wrapped and returned.`

---

*[docs/developer/technical/legend.md, line 438 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMnSiyk7zo4-tM9MsF:-KtMnSiyk7zo4-tM9MsG:b5uv5nt) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L438)):*
> ```Markdown
> This function is used by all layer types to create a flat list of controlled `ProxyWrapper`s. First, controlled wrappers get their `boundingBox` and `query` settings and states disabled. If the controlled layer is not a Dynamic one, layer record is retrieved, its proxy object is obtained, wrapped and returned. Easy.
> 
> If the controlled layer is a Dynamic layer, we first need to wait for it to load since its structure is unknown, then its child tree created (`_createDynamicChildTree`), and their `ProxyWrapper`s extracted as a flat list (their hierarchy is irrelevant since they will never be rendered in UI, so we are content with a flat list).
> ```

`If the controlled layer is a Dynamic layer: we first ...`

---

*[docs/developer/technical/legend.md, line 498 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMoApw1L96B7vNgkU8:-KtMoApw1L96B7vNgkU9:b-qqf65p) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L498)):*
> Feature layers or everything else

I don't think this divide in the legend is explained anywhere.  Maybe explain the limitations of the ESRI v3 API in the Other Notes section above?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Addressed some of the comments.

---

Review status: 2 of 3 files reviewed at latest revision, 22 unresolved discussions.

---

*[docs/developer/technical/legend.md, line 1 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMaw0xIDLge8fYIaQY:-KtMzvduMXZr3DKL6veA:b-oh893w) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L1)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

I would leave a general note somewhere on this page as to what version of the schema was used for all the example code (could be in a comment or somewhere in the footer).  Should make it easier when docs need to be updated to a new version of the schema.
</blockquote></details>

Added to the footer.

---

*[docs/developer/technical/legend.md, line 5 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMN99VDZlnzLjdMfZJ:-KtN--kEpssE2b0c3Kpo:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L5)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

table of contents
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 12 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMNEtvgD29eI7ytQE8:-KtN-5E9N3C08VkT1ik0:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L12)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

formatting loses the line break
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 21 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMN_blNzQTk8OEZOhu:-KtN-JNlxZLn_F2wYQdz:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L21)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

This is repeated from above
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 30 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMNmsE9kV9vX81qr7U:-KtN-aj9tD4KynjeTuWi:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L30)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

remove the `to` after config author
`hide layer` -> `hide layers`
I would rewrite the last sentence as: All this should be used for thematic maps (i.e. maps that tell a certain story with the data).
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 34 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMOzfXqXaipcXM4-gp:-KtN-uGPi6Afk9QrLmLR:b-vgxqk7) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L34)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

`after it's` ->`after it has`
</blockquote></details>

`it was generated` seems more correct to me, no?

---

*[docs/developer/technical/legend.md, line 178 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMbzzMYBMDNoOdoqN0:-KtNG8INDoCkh7PWKTXm:b-z5a3r1) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L178)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

I would give a slightly more detailed example and explain the more important properties (maybe something with layerId, entryIndex, symbologyStack and symbologyRenderStyle).  Something people can easily copy and paste to get started.
</blockquote></details>

Done. See below.

---

*[docs/developer/technical/legend.md, line 224 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMgZVT1g8pc2U7rsgF:-KtNK6Ls9UFoG9Yp0arZ:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L224)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

either `other groups included` or `including other groups`
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 357 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMj-tx2qU9OyIx6IKj:-KtNKEuLp8bsspO8MQ-g:bzgcvtl) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L357)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

This is probably where the user facing side of the docs end.  Maybe add another top level break or have some form of note for the users?
</blockquote></details>

Maybe splitting into two separate files will be better?

---

*[docs/developer/technical/legend.md, line 359 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMjIci9l0UHzwkQvlY:-KtNJ7nuJ81dIb8T7w92:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L359)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

is *currently* 800 lines :)
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 363 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMjljMhLjkknrHXkfW:-KtNJHwlQ0XSQppOywUo:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L363)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

`legend block` -> `legend blocks`
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 365 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMk2091cwaEJtXQGiC:-KtNJLY1pEalXnr6yQ54:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L365)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

`Into` -> `Info`
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 371 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMkKKq8siZpZ6BxxWn:-KtNJPBI2O8kdDM9rvA0:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L371)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

`zero or move` -> `zero or more`
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 432 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMmoc8UJ4qvwrxRuWn:-KtNJUtZ8JVLfQKvOX2I:b-el33dp) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L432)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

`passion` -> `fashion`?
</blockquote></details>

Done. :D

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-:-KtNKQnOBBf00yEhv8fw:b-86rtes)*
<!-- Sent from Reviewable.io -->

alyec commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[docs/developer/technical/legend.md, line 34 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMOzfXqXaipcXM4-gp:-KtNo8UPhBTKLWM8grcd:bg7igkv) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L34)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

`it was generated` seems more correct to me, no?
</blockquote></details>

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

*[docs/developer/technical/legend.md, line 357 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMj-tx2qU9OyIx6IKj:-KtNoUlj_0uKXyaq2klD:bqjvelx) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L357)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Maybe splitting into two separate files will be better?
</blockquote></details>

:white_check_mark: 

---

*[docs/developer/technical/legend.md, line 432 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMmoc8UJ4qvwrxRuWn:-KtNod-P2nF12_1k-5TM:b-cewyyf) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L432)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary><blockquote>

Done. :D
</blockquote></details>

Missed the `h`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
Moved files to the proper places. See here:
https://fgpv-vpgf.github.io/fgpv-vpgf/docs-legend-n-lore/tutorial-legend.html
https://fgpv-vpgf.github.io/fgpv-vpgf/docs-legend-n-lore/tutorial-legend_generation.html

---

Review status: 1 of 5 files reviewed at latest revision, 11 unresolved discussions.

---

*[docs/developer/technical/legend.md, line 87 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMbM93VKoGBt4dukOK:-KtR5hpl_Jrf6HJjdqwU:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L87)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

I would add all four types here, then have the full description sections below (so `There are four types: text, title, image, and unbound layer`).
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 255 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMh_KGMzqQpsG7xboU:-KtRJ3SvYo-PYxzhssT3:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L255)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Maybe provide a bit more guidance to users on when they should use a visibility set vs normal layers (giving a few good and bad use cases).
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 343 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMi9mLEJW9BMby3kPC:-KtRG5kb1ns_Y72K42fe:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L343)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

On the dev side we should explain that trying to catch all these cases would result in a hugely bloated schema that no one would want to ever touch.  A plea for config authors to be sensible seemed much less painful.
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 357 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMj-tx2qU9OyIx6IKj:-KtR5l8rLszk2ZtE9khD:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L357)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

:white_check_mark: 
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 375 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMk_n_yeu4v9MKK73F:-KtRDqljSLT6LRYrG48p:bombmv2) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L375)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

I would make this a sub point of the first bullet since it is covered by that case and just additional info
</blockquote></details>

Good call. Done.

---

*[docs/developer/technical/legend.md, line 379 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMl3Hz1YYGURw4qiLx:-KtRDdTtUVMSqEFYYDm7:bt2ojbw) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L379)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Maybe include a bit about what the proxy object is or a link to the relevant documentation.
</blockquote></details>

Added a short note. Not sure if we have relevant docs yet on proxy objects.

---

*[docs/developer/technical/legend.md, line 432 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMmoc8UJ4qvwrxRuWn:-KtR62odcGzRN5Ym8fap:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L432)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

Missed the `h`
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 436 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMnJuiCDJjbSWqcLCc:-KtRCx7hzkYgd4rnAoA6:b-896fix) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L436)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

`If the controlled layer is not a Dynamic one: the layer record is retrieved, its proxy object is obtained, wrapped and returned.`
</blockquote></details>

Done.

---

*[docs/developer/technical/legend.md, line 438 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMnSiyk7zo4-tM9MsF:-KtRCjNzriMquidAe_Eh:b-qj92pf) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L438)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

`If the controlled layer is a Dynamic layer: we first ...`
</blockquote></details>

Reworded.

---

*[docs/developer/technical/legend.md, line 498 at r2](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-KtMoApw1L96B7vNgkU8:-KtRBXJ3kVrXWwBuh02N:b-257t04) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/3431b6a78cb8f987c784ad6b73d963d847ce0007/docs/developer/technical/legend.md#L498)):*
<details><summary><i>Previously, alyec (Aly Merchant) wrote…</i></summary><blockquote>

> Feature layers or everything else

I don't think this divide in the legend is explained anywhere.  Maybe explain the limitations of the ESRI v3 API in the Other Notes section above?
</blockquote></details>

Added some notes on that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-:-KtRMvV7nwoxipZfFFZK:bnylcjx)*
<!-- Sent from Reviewable.io -->

alyec commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355#-:-KtS0O0Tzl6AqGWXxtEe:bnfp4nl)*
<!-- Sent from Reviewable.io -->

james-rae commented:
Reviewed 1 of 1 files at r3, 4 of 4 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/2355)*
<!-- Sent from Reviewable.io -->
